src/chrono/util.cljc

```diff
@@ -24,10 +26,23 @@
 
 (def iso-fmt [:year "-" :month "-" :day "T" :hour ":" :min ":" :sec "." :ms])
 
+(defn parse-name [name {lang :lang unit :unit}]
+  (-> (locale lang)
+      (get unit)
+      (->> (filter #(re-matches (-> % val :regex re-pattern) name)))
+      doall
+      ffirst))
+
 (defn parse-int [x]
   (when (string? x)
-    #?(:clj (Integer/parseInt x)
-       :cljs (js/parseInt  x))))
+    #?(:clj (try (Integer/parseInt x) (catch NumberFormatException e nil))
+       :cljs (js/parseInt x))))
+
+(defn parse-val [x f]
+  (or (parse-int x)
+      (parse-name x f)
+      (throw
+       (Exception. (str "Can't parse string \"" x "\" with format " f)))))
```

Comment:
Better to return nil instead of throwing an exception. And in the strict parse return nil if couldn't parse whole string

Reply:
Done in ab528be6e9645a40111434b5e830f06586533c4d
Also fixed logic while parse returns nil if can parse only part of fields from format vector, added [test](https://github.com/Flawless/chrono/blob/24d15270d0748f522eb92490309c16651722a97f/test/chrono/io_test.clj#L117) for this case

Reply:
Tests for similar logic for `strict-parse` presented [here](https://github.com/Flawless/chrono/blob/be5a4fff23f38f673a3df5612b121981554291f4/test/chrono/io_test.clj#L132)